docs/ai/conceptual/understanding-tokens.md

```diff
@@ -0,0 +1,118 @@
+---
+title: "Understanding tokens"
+description: "Understand how large language models (LLMs) use tokens to analyze semantic relationships and generate natural language outputs"
+author: haywoodsloan
+ms.topic: concept-article
+ms.date: 04/12/2024
+
+#customer intent: As a .NET developer, I want understand how large language models (LLMs) use tokens so I can add semantic analysis and text generation capabilities to my .NET projects.
+
+---
+
+# Understanding tokens
+
+Tokens are the words or character sets that large language models (LLMs) decompose text into. During training, an LLM tokenizes the training text then analyzes the semantic relationships between these tokens, such as how commonly they're used together or if they're used in similar contexts. Once trained, an LLM uses those patterns and relationships to generate a sequence of output tokens based on the input sequence.
+
+## Turning text into tokens
+
+The first step when training an LLM or inputting a prompt is tokenization. The model breaks down the text into a set of strings, known as _tokens_. These tokens can be distinct words, character sets, or combinations of words and punctuation. The set of unique tokens a model has been trained on is referred to as its _vocabulary_.
+
+For example, consider the following sentence:
+
+> I heard a dog bark loudly at a cat
```

Comment:
```suggestion
> I heard a dog bark loudly at a cat.
```

Reply:
I want to ommit the '.' to avoid discussing how punctuation is handeled in word-based tokenization (in reality there's a few different approaches to handling punctuation) to keep the focus of this example more straightfoward.